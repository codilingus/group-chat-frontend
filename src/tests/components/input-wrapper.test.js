import React from 'react';
import { shallow } from 'enzyme';
import InputWrapper from '../../components/input-wrapper';

test('should render InputWrapper component correctly', () => {
  const wrapper = shallow(<InputWrapper />);
  expect(wrapper).toMatchSnapshot();
});

// test('should set value on input change', () => {
//   const onChange = jest.fn();
//   const value = 'some value';
//   const wrapper = shallow(<InputWrapper />);
//   wrapper.find('input').simulate('change', {
//     target: { value }
//   });
//   expect(wrapper.state('value')).toBe(value);
//   expect(onChange).toHaveBeenLastCalledWith({ value })
// });
