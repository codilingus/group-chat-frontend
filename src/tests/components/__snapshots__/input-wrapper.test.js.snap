// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputWrapper component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputWrapper />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span />,
        <input
          className="input-wrapper-input"
          onChange={[Function]}
        />,
      ],
      "className": "input-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-wrapper-input",
          "onChange": [Function],
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span />,
          <input
            className="input-wrapper-input"
            onChange={[Function]}
          />,
        ],
        "className": "input-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-wrapper-input",
            "onChange": [Function],
            "placeholder": undefined,
            "type": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
